tls/s2n_resume.c

```diff
@@ -611,25 +628,50 @@ int s2n_config_wipe_expired_ticket_crypto_keys(struct s2n_config *config, int8_t
     return 0;
 }
 
+int s2n_verify_unique_ticket_key_comparator(void *a, void *b)
+{
+    return (memcmp((uint8_t *) a, (uint8_t *) b, SHA_DIGEST_LENGTH));
+}
+
 int s2n_verify_unique_ticket_key(struct s2n_config *config, uint8_t *hash, uint16_t *insert_index)
 {
-    int low = 0;
-    int top = config->ticket_key_hashes->num_of_elements - 1;
-
-    /* Binary search */
-    while (low <= top) {
-        int mid = low + ((top - low) / 2);
-        int m = memcmp((uint8_t *) s2n_array_get(config->ticket_key_hashes, mid), hash, SHA_DIGEST_LENGTH);
-
-        if (m == 0) {
-            return -1;
-        } else if (m > 0) {
-            top = mid - 1;
-        } else if (m < 0) {
-            low = mid + 1;
-        }
+    int result = s2n_array_binary_search(0,
+                                         config->ticket_key_hashes->num_of_elements - 1,
+                                         config->ticket_key_hashes,
+                                         hash,
+                                         s2n_verify_unique_ticket_key_comparator);
+
+    if (result == -1) {
+        return result;
     }
 
-    *insert_index = low;
+    *insert_index = result;
     return 0;
 }
+
+int s2n_config_store_ticket_key_comparator(void *a, void *b)
+{
+    if (((struct s2n_ticket_key *) a)->intro_timestamp >= ((struct s2n_ticket_key *) b)->intro_timestamp)
+        return 1;
```

Comment:
style: include `{}` after if and else

Reply:
also return 0 if they're equal?

Reply:
> also return 0 if they're equal?

0 will cause  s2n_array_binary_search to return -1. Hence did >=, so that an index is returned.

```
        if (m == 0) {
            /* Return -1 when a match is found */
            return -1;
```